Microsoft.Toolkit.Uwp.SampleApp/Controls/AadAuthControl.xaml.cs

```diff
@@ -0,0 +1,139 @@
+            typeof(AadAuthControl),
+            new PropertyMetadata(Visibility.Visible));
+
+        public Visibility IsShowSignInButton
+        {
+            get { return (Visibility)GetValue(IsShowSignInButtonProperty); }
+            set { SetValue(IsShowSignInButtonProperty, value); }
+        }
+
+        public static readonly DependencyProperty IsEnableSignInButtonProperty = DependencyProperty.Register(
+            nameof(IsEnableSignInButton),
+            typeof(bool),
+            typeof(AadAuthControl),
+            new PropertyMetadata(false));
+
+        public bool IsEnableSignInButton
+        {
+            get { return (bool)GetValue(IsEnableSignInButtonProperty); }
+            set { SetValue(IsEnableSignInButtonProperty, value); }
+        }
+
+        private AadAuthenticationManager _aadAuthenticationManager = AadAuthenticationManager.Instance;
+
+        public AadAuthControl()
+        {
+            InitializeComponent();
+
+            _aadAuthenticationManager.PropertyChanged += AadAuthenticationManager_PropertyChanged;
+            ClientId.Text = _aadAuthenticationManager.ClientId ?? string.Empty;
+
+            ClientId.TextChanged += (object sender, TextChangedEventArgs e) =>
+            {
+                if (string.IsNullOrEmpty(ClientId.Text.Trim()))
+                {
+                    IsEnableSignInButton = false;
+                    return;
+                }
+
+                _aadAuthenticationManager.Initialize(
+                    ClientId.Text.Trim(),
+                    AadLogin.RequiredDelegatedPermissions
+                        .Concat(ProfileCard.RequiredDelegatedPermissions)
+                        .Concat(PeoplePicker.RequiredDelegatedPermissions)
+                        .Concat(SharePointFileList.RequiredDelegatedPermissions)
+                        .ToArray());
+
+                IsEnableSignInButton = true;
+            };
+
+            Scopes.Text = string.Join(", ", AadLogin.RequiredDelegatedPermissions
+                .Union(ProfileCard.RequiredDelegatedPermissions)
+                .Union(PeoplePicker.RequiredDelegatedPermissions)
+                .Union(SharePointFileList.RequiredDelegatedPermissions)
+                .Distinct());
+
+            IsEnableSignInButton = ClientId.Text.Trim().Length > 0;
+            RereshControlVisibility();
+        }
+
+        private void AadAuthenticationManager_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(_aadAuthenticationManager.IsAuthenticated))
+            {
+                RereshControlVisibility();
+
+                if (_aadAuthenticationManager.IsAuthenticated)
+                {
+                    ClientId.IsReadOnly = true;
+                }
+                else
+                {
+                    ClientId.IsReadOnly = false;
+                }
+            }
+        }
+
+        private void RereshControlVisibility()
```

Comment:
This can be done in XAML.  IsAuthenticated will notify itself of changes, so you can bind the boolean to visibility directly in XAML.  (then you won't have to fix the typo in the function name either) ;)

Reply:
Done and committed, the function has been removed.